cmd/skaffold/app/skaffold_test.go

```diff
@@ -0,0 +1,41 @@
+/*
+Copyright 2019 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	"github.com/GoogleContainerTools/skaffold/testutil"
+)
+
+func TestMain(t *testing.T) {
+	var (
+		output    bytes.Buffer
+		errOutput bytes.Buffer
+	)
+
+	defer func(args []string) { os.Args = args }(os.Args)
+	os.Args = []string{"skaffold", "help"}
+
+	err := Run(&output, &errOutput)
+
+	testutil.CheckError(t, false, err)
+	testutil.CheckContains(t, "Available Commands", output.String())
+	testutil.CheckDeepEqual(t, "", errOutput.String())
+}
```

Comment:
Can we also have a test case for invalid command and see if error is returned.

Reply:
Added!

Reply:
@tejal29 woot! Discovered a race condition thanks to this test